vslib/src/MACsecFilter.cpp

```diff
@@ -30,6 +32,17 @@ void MACsecFilter::enable_macsec_device(
     SWSS_LOG_ENTER();
 
     m_macsecDeviceEnable = enable;
+
+    // The function, execute(), may be running in another thread,
+    // the macsec device enable state cannot be changed in the busy state.
+    // Because if the macsec device was deleted in the busy state,
+    // the error value of function, forward, will be returned
+    // to the caller of execute()
+    // and the caller thread will exit due to the error return value.
+    while(get_state() == MACsecFilter::MACsecFilterState::MACSEC_FILTER_STATE_BUSY)
```

Comment:
this is just empty loop since get_state return just state, so it's like a spin lock using 100% cpu, how long this while will potentially spin?

Reply:
It will only be locked during the below block once, if you don't like a pure spin lock using 100% cpu, I can add a tiny sleep about 1 ms.
https://github.com/Azure/sonic-sairedis/blob/276eb3506f3b93ce41436134448a241e9aa5187d/vslib/src/MACsecEgressFilter.cpp#L19-L50